internal/tools/aws/rds.go

```diff
@@ -0,0 +1,193 @@
+		dbSecurityGroups = append(dbSecurityGroups, *sg.GroupId)
+	}
+
+	if len(dbSecurityGroups) == 0 {
+		return []string{}, fmt.Errorf("unable to find security groups tagged for Mattermost DB usage: %s=%s", DefaultDBSecurityGroupTagKey, DefaultDBSecurityGroupTagValue)
+	}
+
+	logger.WithField("security-group-ids", dbSecurityGroups).Debugf("Found %d DB tagged security groups", len(dbSecurityGroups))
+
+	return dbSecurityGroups, nil
+}
+
+func (a *Client) rdsEnsureDBClusterCreated(awsID, username, password string, logger log.FieldLogger) error {
+	svc := rds.New(session.New(), &aws.Config{
+		Region: aws.String(DefaultAWSRegion),
+	})
+
+	_, err := svc.DescribeDBClusters(&rds.DescribeDBClustersInput{
+		DBClusterIdentifier: aws.String(awsID),
+	})
+	if err == nil {
+		logger.WithField("db-cluster-name", awsID).Debug("AWS DB cluster already created")
+
+		return nil
+	}
+
+	dbSecurityGroupIDs, err := a.rdsGetDBSecurityGroupIDs(logger)
+	if err != nil {
+		return err
+	}
+
+	input := &rds.CreateDBClusterInput{
+		AvailabilityZones: []*string{
+			aws.String("us-east-1a"),
+			aws.String("us-east-1b"),
+			aws.String("us-east-1c"),
+		},
+		BackupRetentionPeriod: aws.Int64(7),
+		DBClusterIdentifier:   aws.String(awsID),
+		DatabaseName:          aws.String("mattermost"),
+		EngineMode:            aws.String("provisioned"),
+		Engine:                aws.String("aurora-mysql"),
+		EngineVersion:         aws.String("5.7"),
+		MasterUserPassword:    aws.String(password),
+		MasterUsername:        aws.String(username),
+		Port:                  aws.Int64(3306),
+		StorageEncrypted:      aws.Bool(false),
+		DBSubnetGroupName:     aws.String(DefaultDBSubnetGroupName),
+		VpcSecurityGroupIds:   aws.StringSlice(dbSecurityGroupIDs),
+	}
+
+	_, err = svc.CreateDBCluster(input)
+	if err != nil {
+		return err
+	}
+
+	logger.WithField("db-cluster-name", awsID).Debug("AWS DB cluster created")
+
+	return nil
+}
+
+func (a *Client) rdsEnsureDBClusterInstanceCreated(awsID, instanceName string, logger log.FieldLogger) error {
+	svc := rds.New(session.New(), &aws.Config{
+		Region: aws.String(DefaultAWSRegion),
+	})
+
+	_, err := svc.DescribeDBInstances(&rds.DescribeDBInstancesInput{
+		DBInstanceIdentifier: aws.String(instanceName),
+	})
+	if err == nil {
+		logger.WithField("db-instance-name", instanceName).Debug("AWS DB instance already created")
+
+		return nil
+	}
+
+	_, err = svc.CreateDBInstance(&rds.CreateDBInstanceInput{
+		DBClusterIdentifier:  aws.String(awsID),
+		DBInstanceIdentifier: aws.String(instanceName),
+		DBInstanceClass:      aws.String("db.r5.large"),
+		Engine:               aws.String("aurora-mysql"),
+		PubliclyAccessible:   aws.Bool(false),
+	})
+	if err != nil {
+		return err
+	}
+
+	logger.WithField("db-instance-name", instanceName).Debug("AWS DB instance created")
+
+	return nil
+}
+
+func rdsGetDBCluster(awsID string, logger log.FieldLogger) (*rds.DBCluster, error) {
+	svc := rds.New(session.New(), &aws.Config{
```

Comment:
I'm seeing this pattern in a number of places, maybe these three lines should be in a helper function?

Reply:
So, the client created here is often slightly different in that it is specific to the AWS service being used. Note that it's `rds` here. That said, we could make a helper function for each AWS service. Do you think that warrants the extra code?